src/Compilers/CSharp/Portable/Binder/SwitchBinder.cs

```diff
@@ -205,7 +205,8 @@ private BoundExpression ConvertCaseExpression(TypeSymbol switchGoverningType, CS
 
             if (switchGoverningType.IsNullableType() && convertedCaseExpression.Kind == BoundKind.Conversion)
             {
-                constantValueOpt = ((BoundConversion)convertedCaseExpression).Operand.ConstantValue;
+                var operand = ((BoundConversion)convertedCaseExpression).Operand;
+                constantValueOpt = CreateConversion(operand, switchGoverningType.GetNullableUnderlyingType(), diagnostics).ConstantValue;
```

Comment:
We are probably never interested in errors reported by CreateConversion here, I suggest suppressing them by passing new diagnostics bag to the method.